api/wb/views.py

```diff
@@ -0,0 +1,95 @@
+from rest_framework import status
+from rest_framework.exceptions import NotFound
+from rest_framework.response import Response
+
+from osf.models import AbstractNode
+from rest_framework.views import APIView
+from addons.osfstorage.models import OsfStorageFileNode, OsfStorageFolder
+from api.base.utils import get_object_or_error
+from api.base.parsers import HMACSignedParser
+from api.wb.serializers import (
+    WaterbutlerMetadataSerializer
+)
+
+class FileMetadataView(APIView):
+    """
+    Mixin with common code for WB move/copy hooks
+    """
+    parser_classes = (HMACSignedParser,)
+    serializer_class = WaterbutlerMetadataSerializer
+    view_category = 'wb'
+    node_lookup_url_kwarg = 'node_id'
+
+    def get_object(self):
+        return self.get_node(self.kwargs[self.node_lookup_url_kwarg])
+
+    def get_node(self, node_id):
+        node = get_object_or_error(
+            AbstractNode,
+            node_id,
+            self.request,
+            display_name='node'
+        )
+        if node.is_registration:
```

Comment:
Why was this check added? I don't see it in the original v1 endpoints.

Reply:
[osfstorage_move_hook](https://github.com/CenterForOpenScience/osf.io/blob/develop/addons/osfstorage/views.py#L128) has a waterbutler_opt_hook decorator on it - [there's a must_not_be_registration check there](https://github.com/CenterForOpenScience/osf.io/blob/develop/addons/osfstorage/decorators.py#L66)

Reply:
Ah ok, I missed that. Thanks.

Reply:
Looks like the decorator checks archiving status:

```python
        if node.is_registration and not node.archiving:
            raise HTTPError(
                http.BAD_REQUEST,
                data={
                    'message_short': 'Registrations cannot be changed',
                    'message_long': "The operation you're trying to do cannot be applied to registered projects, which are not allowed to be changed",
                }
            )
```

Also, a 400 response is returned instead of a 404. It may make sense to keep it as a 400. That would allow the client (WB) too differentiate between request for an invalid node GUID and a request for an archived registration.